buildSrc/src/main/java/io/bitrise/trace/internal/UpdateChangeLogTask.java

```diff
@@ -0,0 +1,482 @@
+        static final Pattern footerPattern = Pattern.compile(footerRegex);
+
+        /**
+         * Constructor for class.
+         *
+         * @param logger a {@link Logger} that will provide log outputs to the console.
+         */
+        public ChangeLogHelper(final Logger logger) {
+            this.logger = logger;
+        }
+
+        /**
+         * Gets the name of the given release. Determines the new version from the change log entries. This is
+         * concatenated
+         * with the current date.
+         *
+         * @param lastTag the previous tag.
+         * @return the name of the release.
+         */
+        String getReleaseName(final Ref lastTag, final List<ChangeLogEntry> changeLogEntries) {
+            final String previousTagShortName = lastTag.getName().substring(Constants.R_TAGS.length());
+            logger.debug("The name of the last tag was \"{}\"", previousTagShortName);
+            final Set<String> entryTypeSet =
+                    changeLogEntries.stream().map(ChangeLogEntry::getType).collect(Collectors.toSet());
+            entryTypeSet.forEach(type -> logger.debug("Found in new commits type \"{}\"", type));
+
+            return String.format("# %s - %s", getNewVersion(previousTagShortName, entryTypeSet), getCurrentDate());
+        }
+
+        /**
+         * Determines the new version from the change log entries (patch, minor or major release). If there are no
+         * changes, it will increase the patch version.
+         *
+         * @param version the previous version.
+         * @param typeSet the Set of types of the change log entries.
+         * @return the new version.
+         */
+        String getNewVersion(final String version, final Set<String> typeSet) {
+            final String[] versionNumbers = version.split("\\.");
+            if (typeSet.stream().anyMatch(majorCommitTypes::contains)) {
+                logger.debug("New version will have major version increase!");
+                versionNumbers[0] = String.valueOf(Integer.parseInt(versionNumbers[0]) + 1);
+                versionNumbers[1] = "0";
+                versionNumbers[2] = "0";
+            } else if (typeSet.stream().anyMatch(minorCommitTypes::contains)) {
+                logger.debug("New version will have minor version increase!");
+                versionNumbers[1] = String.valueOf(Integer.parseInt(versionNumbers[1]) + 1);
+                versionNumbers[2] = "0";
+            } else {
+                logger.debug("New version will have patch version increase!");
+                versionNumbers[2] = String.valueOf(Integer.parseInt(versionNumbers[2]) + 1);
+            }
+            return String.format("%s.%s.%s", versionNumbers[0], versionNumbers[1], versionNumbers[2]);
+        }
+
+        /**
+         * Gets the current date in a YYYY-MM-DD format.
+         *
+         * @return the String value of the current date.
+         */
+        private String getCurrentDate() {
```

Comment:
not sure if we should have a consensus on the format of the date - the changelog for the trace-sdk looks like https://github.com/bitrise-io/trace-android-sdk/blob/main/CHANGELOG.md e.g. - 11th Feb 2021 - perhaps sticking with mm-dd-yyyy (and we'd need to update trace-sdk Feb to be 02)?

Reply:
You are right, we haven't discussed this yet, happy to have a chat about it. I used the [ISO](https://en.wikipedia.org/wiki/ISO_8601) format, because that is the international standard format, but we should definitely keep this consistent across different products (even among teams). I think we should start a conversation about this with the team, but I am interested in your opinion, please let me know. 🙂

Reply:
i agree with the international ISO8601 format too - i'm just not sure I like the dots 😅 Would you be adverse to hyphens e.g. 2021-02-24 instead of 2021.02.24?

Reply:
Sure, updated it, nice catch!